cls/SourceControl/Git/Extension.cls

```diff
@@ -49,7 +49,11 @@ Method UserAction(Type As %Integer, Name As %String, InternalName As %String, Se
     if Type = 1, Name = 5 {
         // reroute to Status menu option
         set Name = "%SourceMenu,Status"
-    } 
+        if (settings.gitUserName = "") || (settings.gitUserEmail = "") {
+            set Target = "Warning: Git committer name or email is not configured. Go to settings."
+            set Action = 6
```

Comment:
We want to take this same approach not just on initial connection but also on the Commit and Sync menu items. If attribution settings aren't set up, in any of these cases, we want to change Action to 6 (show an alert), put the same helpful message in Target, and not try to Commit or Sync (rather than trying to and failing later in the process). This is especially relevant for Sync.

If you want to refactor this to a helper method, that's great. It's also good to have the detection in RunGitCommandWithInput specifically, because we also want to be helpful for commits via the Git WebUI.